### PR TITLE
add memory pair game task

### PR DESCRIPTION
**Pull request by frostwolm**
### Memory – Pair Game

[Demo](https://frostwolm.github.io/memory-pair-game/) |
[Code base](https://github.com/frostwolm/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/frostwolm/pair-game/js/app.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+let clickedCards = [];
+let cardsNum = 12;
+let imgIDsArray = [];
+let _checkNow = false;
+
+init(cardsNum);
```
Comment (AMashoshyna) [inline submissions/frostwolm/pair-game/js/app.js]: Function "init" has no formal parameters

File: submissions/frostwolm/pair-game/js/app.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+function init(){
+    document.querySelector('.game-container').innerHTML = '';
```
Comment (AMashoshyna) [inline submissions/frostwolm/pair-game/js/app.js]: You make this query several times. DOM queries are expensive in terms of performance and should not be repeated if the element is not expected to change. Please save reference to DOM node in a variable and refer to it.

File: submissions/frostwolm/pair-game/js/app.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+  }
+  if (clickedCards.length == 2) {
+    _checkNow = true;
+    setTimeout(compareCards, 500);
+  }
+}
+
+function compareCards() {
+  if (clickedCards[0].parentNode.parentNode.parentNode.dataset.imgid ==
```
Comment (AMashoshyna) [inline submissions/frostwolm/pair-game/js/app.js]: Relying in the nesting order of HTML elements is very fragile: if your layout changes due to redesign, your code will break. Please find a way to have controlled identifiers (data* attributes, ids, class names).

**Comment by AMashoshyna** [comment]
Very nice job! 
Some bugs I found:
- when the cards do not match, the turn with different speed
- the game does not restart. If I have won and start a new game without reloading the page, the game alerts "Win" every time a turn new card pair

image


As for code - please see my comments

**Comment by AMashoshyna** [comment]
Sorry for taking so long with the answer - I had to do some research.
So, the best solution I could come up with is to move `setTimeout` inside `forEach` callback like this:
```javascript
    clickedCards.forEach(cardElem => {
      setTimeout(() => flip(cardElem), 100);
    })
```
Pay attention to arrow function inside `setTimeout`: callback passed to `setTimeout` should be a function, not an expression.
I will ask other mentors for better ideas and let you know if we find something.

**Comment by frostwolm** [comment]
Done

**Comment by AMashoshyna** [comment]
Well done!

**Pull request merged**